File: app/src/main/java/com/xch/im/imtest/MainActivity.java
package com.xch.im.imtest;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    OvalToCircle mButton;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mButton = (OvalToCircle) findViewById(R.id.animation_btn);
        mButton.setAnimationButtonListener(new OvalToCircle.AnimationButtonListener() {
            @Override
            public void onClickListener() {
                mButton.start();
            }

            @Override
            public void animationFinish() {
                mButton.reset();
            }
        });
    }
}

File: app/src/main/java/com/xch/im/imtest/activity/MainActivity.kt
package com.xch.im.imtest.activity

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import com.xch.im.imtest.R
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity(), View.OnClickListener {
    private var mBtn_button: Button? = null


    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.main_btn_button -> startActivity(Intent(this, ButtonViewActivity::class.java))
        }
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initListener()

    }


    private fun initListener() {
        main_btn_button.setOnClickListener(this)
    }
}


